pyscript.core/esm/runtime/_python.js

```diff
@@ -0,0 +1,27 @@
+import { clean, writeFile as writeFileUtil } from "./_utils.js";
+
+// REQUIRES INTEGRATION TEST
+/* c8 ignore start */
+export const run = (runtime, code) => runtime.runPython(clean(code));
+
+export const runAsync = (runtime, code) => runtime.runPythonAsync(clean(code));
+
+export function runEvent(runtime, code, key) {
+    code = `import js;event=js.__events.get(${key});${code}`;
+    return this.run(runtime, code);
+}
+
+const worker = (method) =>
+    function (runtime, code, xworker) {
+        code = `from js import xworker;${code}`;
+        globalThis.xworker = xworker;
+        return this[method](runtime, code);
```

Comment:
`from xworker import xworker` this is ugly to me but I think this `pyodide.globals.set("xworker", xworker)` is aligned with what wasmoon does and it might fix the previous *event* case too ... not sure it works in MicroPython though so ... to answer this:

> This is another messy use of a global

I fully agreed but I need a way that is consistent across runtimes and it actually works ... Damien wasn't happy neither about the global usage of xworker but it's needed to make the logic work as it does now.

Happy to improve but surely not over an already merged PR :-)

Reply:
> `from xworker import xworker` this is ugly to me

It's pretty standard looking Python. E.g `from datetime important datetime`, `from dis import dis`, `from print import pprint` etc etc.

Reply:
it's hard for me to reflect that on other runtimes though ... and we're talking about a worker scope, not a global page pollution where any 3rd party library might have a say there ... is the global.set('xworker', xworker) that bad compared to this? it simplifies documentation too.

Reply:
No, I think based on what you are saying `global.set('xworker', xworker)` sounds like the way to go.

Reply:
I'll check MicroPython and eventually diverge in there with bootstrap as that's expected to both work and be possible (the different runtime bootstrap dance, which is already the case, see the Promise polyfill issue in MicroPython).